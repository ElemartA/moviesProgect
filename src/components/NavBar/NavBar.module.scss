.header__nav {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 100%;
  margin-bottom: 70px;
  padding: 0;
  .header__list {
    max-width: 1200px;
    display: flex;
    flex-wrap: wrap;
    .header__link {
      margin-right: 30px;
      text-decoration: none;
      font-size: 24px;
      color: black;
      font-weight: bold;
    }
    .header__link:hover {
      color: white;
      transition-duration: 0.4s;
    }
    li:last-child a {
      margin-right: 0;
    }
  }
}
