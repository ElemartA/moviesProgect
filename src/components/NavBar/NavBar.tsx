import React from "react";
import { Link } from "react-router-dom";
import s from "./NavBar.module.scss";

class NavBar extends React.Component {
  render() {
    return (
      <div>
        <nav>
          <ul className={s.header__nav}>
            <li className={s.header__list}>
              <Link data-testid="main" to="" className={s.header__link}>
                Главная
              </Link>
            </li>
            <li className={s.header__list}>
              <Link data-testid="about" to="/about" className={s.header__link}>
                О нас
              </Link>
            </li>
            <li className={s.header__list}>
              <Link data-testid="form" to="/form" className={s.header__link}>
                Форма
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    );
  }
}

export default NavBar;
