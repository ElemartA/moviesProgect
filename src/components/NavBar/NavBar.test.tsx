import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NavBar from "./NavBar";

describe("NavBar", () => {
  it("renders NavBar component", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    screen.debug();
    expect(screen.getByRole("navigation")).toBeInTheDocument();
    expect(screen.queryByText(/Главная/i)).toBeInTheDocument();
    expect(screen.queryByText(/О нас/i)).toBeInTheDocument();
    expect(screen.queryByText(/Форма/i)).toBeInTheDocument();
  });
});
