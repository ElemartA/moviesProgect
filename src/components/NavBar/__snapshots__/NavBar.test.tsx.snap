// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders NavBar component 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <ul
        class="header__nav"
      >
        <li
          class="header__list"
        >
          <a
            class="header__link"
            data-testid="main"
            href="/"
          >
            Главная
          </a>
        </li>
        <li
          class="header__list"
        >
          <a
            class="header__link"
            data-testid="about"
            href="/about"
          >
            О нас
          </a>
        </li>
        <li
          class="header__list"
        >
          <a
            class="header__link"
            data-testid="form"
            href="/form"
          >
            Форма
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
